### PR TITLE
Auto handle Apple char update failures

### PR DESCRIPTION
**Pull request by rohitsangwan01**

### CODE REVIEW

File: example/lib/home_controller.dart [hunk 1]
```diff
@@ -70,17 +83,27 @@ class HomeController extends GetxController {
...
+            deviceId: deviceId,
+            subscribedChars: {characteristicId},
+          ));
         }
       } else {
-        devices.removeWhere((element) => element == deviceName);
+        if (index != -1) {
+          devices[index].subscribedChars.remove(characteristicId);
+          if (devices[index].subscribedChars.isEmpty) {
+            devices.remove(index);
```
Comment (fotiDim) [inline example/lib/home_controller.dart]:
```suggestion
            devices.removeAll { $0.identifier == central.identifier }

```

File: darwin/Classes/BlePeripheralPlugin.swift [hunk 2]
```diff
@@ -109,21 +111,46 @@ private class BlePeripheralDarwin: NSObject, BlePeripheralChannel, CBPeripheralM
...
-            peripheralManager.updateValue(value.toData(), for: char!, onSubscribedCentrals: [centralDevice!])
-        } else {
-            peripheralManager.updateValue(value.toData(), for: char!, onSubscribedCentrals: nil)
         }
+        isCharUpdateProcessing = false
+    }
+
+    func peripheralManagerIsReady(toUpdateSubscribers _: CBPeripheralManager) {
+        // print("Ready to update char")
+        processCharUpdateQueue()
```
Comment (fotiDim) [inline darwin/Classes/BlePeripheralPlugin.swift:153]: @rohitsangwan01 how about this implementation?

```swift
private let maxQueueSize = 1000 // Set a reasonable maximum queue size

private func processCharUpdateQueue() {
    guard !isCharUpdateProcessing, !charUpdateQueue.isEmpty else {
        return
    }
    
    isCharUpdateProcessing = true
    var retryCount = 0
    let maxRetries = 3
    
    while !charUpdateQueue.isEmpty {
        // Drop older updates if queue gets too large
        if charUpdateQueue.count > maxQueueSize {
            charUpdateQueue = Array(charUpdateQueue.suffix(maxQueueSize))
            continue
        }
        
        guard let charUpdate = charUpdateQueue.first else { break }
        
        let success = peripheralManager.updateValue(
            charUpdate.data,
            for: charUpdate.characteristic,
            onSubscribedCentrals: charUpdate.central.map { [$0] }
        )
        
        if success {
            charUpdateQueue.removeFirst()
            retryCount = 0
        } else {
            retryCount += 1
            if retryCount >= maxRetries {
                // Remove failed update after max retries
                charUpdateQueue.removeFirst()
                retryCount = 0
            }
            break
        }
    }
    
    isCharUpdateProcessing = false
}
```

**Review by fotiDim — COMMENTED** [review]
I tested it and it works. I made 2 comment suggestions, inspired from the AI review.

Also another thing to consider is using `OperationQueue`. I made a commit to use this instead. It seems to work. If you have any concerns we can revert it or put it in another PR.

**Comment by rohitsangwan01** [comment]
@fotiDim first suggestion from AI was to use OpeartionQueue in swift, and i tried that as well, it felt over complicated for our usecase, as we dont really need this much complex implementation
Thats why i came up with that solution of using a siimple list to process queue, and because Swift auto notify us if we are ready to process next updates, we dont really need our own implementation of managing queue

i still feel that your last commit complicated things, because we dont need retry mechanism, like i said, swift will auto notify us that we are ready to update chars

also i saw that now we are using Thread.sleep which might cause lag in ui ( not sure )

```swift
Thread.sleep(forTimeInterval: 0.1 * Double(retryCount))
```

but anyway i would say we don't really have to overcomplicate this with `OperationQueue` and any kind of retry mechanism

**Comment by fotiDim** [comment]
@rohitsangwan01 I reverted my commit. Please check the comment I left for your code.

**Pull request merged**